src/Files.App/DataModels/NavigationControlItems/LocationItem.cs

```diff
@@ -52,11 +54,38 @@ public bool IsExpanded
 		}
 
 		public bool IsInvalid { get; set; } = false;
-
+		public bool IsRecycleBin => string.Equals(Path, "shell:RecycleBinFolder", StringComparison.OrdinalIgnoreCase);
 		public SectionType Section { get; set; }
 
 		public ContextMenuOptions MenuOptions { get; set; }
 
 		public int CompareTo(INavigationControlItem other) => Text.CompareTo(other.Text);
 	}
-}
\ No newline at end of file
+
+	public class RecycleBinLocationItem : LocationItem
+	{
+		public void RefreshSpaceUsed()
+		{
+			SpaceUsed = RecycleBinHelpers.GetSize();
+		}
+
+		private ulong spaceUsed;
+		public ulong SpaceUsed
+		{
+			get => spaceUsed;
+			set { 
+				if (SetProperty(ref spaceUsed, value))
+					OnPropertyChanged(nameof(ToolTipText));
+
+				App.Logger.Warn(ToolTipText + $", real value: {spaceUsed}");
+			}
+		}
+
+		public override string ToolTipText => "RecycleBinSizeText".GetLocalizedResource() + SpaceUsed.ToSizeString();
+
+		public RecycleBinLocationItem()
+		{
+			RecycleBinHelpers.sidebarBin = this;
```

Comment:
not really a fan of holding a reference like this, and fortunately I don't think it's needed 

either the `SidebarPinnedModel` or the `RecycleBinLocationItem` should just subscribe to the events in the watcher -- you can also create a new event that fires on any change to avoid subscribing to three.

it also allows removing the side effect in the watcher, the watcher shouldn't do anything aside from emitting events